lib/api/CDataFrameAnalyzer.cc

```diff
@@ -247,12 +254,32 @@ void CDataFrameAnalyzer::addRowToDataFrame(const TStrVec& fieldValues) {
     });
 }
 
-void CDataFrameAnalyzer::writeResultsOf(const CDataFrameAnalysisRunner& analysis) const {
-    // TODO Revisit this can probably be core::CRapidJsonLineWriter.
-    // TODO This should probably be a member variable so it is only created once.
-    auto outStream = m_OutStreamSupplier();
-    core::CRapidJsonConcurrentLineWriter outputWriter{*outStream};
+void CDataFrameAnalyzer::reportProgress(const CDataFrameAnalysisRunner& analysis,
+                                        core::CRapidJsonConcurrentLineWriter& writer) const {
+    double progress{0.0};
+    while (analysis.finished() == false) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        double latestProgress{analysis.progress()};
+        if (latestProgress > progress) {
```

Comment:
This could end up with very fine grained progress reporting for long running analyses. Imagine if an analysis takes 1hr reporting every 100ms you're going to get 36000 progress reports in increments of 0.003% of complete depending on the granularity of the progress monitoring in low level code. I'd make this something like if `latestProgress > progress + 0.01`.